src/Microsoft.Health.Fhir.Core/Features/Search/SearchValues/ReferenceSearchValueParser.cs

```diff
@@ -46,7 +47,11 @@ public ReferenceSearchValue Parse(string s)
             if (match.Success)
             {
                 string resourceTypeInString = match.Groups[ResourceTypeCapture].Value;
-                ResourceType resourceType = Enum.Parse<ResourceType>(resourceTypeInString);
+
+                if (!ModelFactory.IsKnownResource(resourceTypeInString))
+                {
+                    throw new RequestNotValidException(string.Format(Resources.ResourceNotSupported, resourceTypeInString));
```

Comment:
I think maybe `ArgumentException` is more appropriate since Parse could be used by other calls too.